org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchConfigurationWorkingCopy.java

```diff
@@ -271,16 +275,33 @@ private void doSave0(IProgressMonitor monitor) throws CoreException {
 		}
 	}
 
+	private static String getLineDelimiterPreference(IFile file) {
```

Comment:
Calling this allows you to retrieve the newline preference (from the appropriate scope). If none is specified in any scope (project/workspace) then we default to the system line separator (what currently happens).